Sources/Rendering/Core/ColorTransferFunction/index.js

```diff
@@ -494,6 +494,24 @@ function vtkColorTransferFunction(publicAPI, model) {
         x = 0.5 * (xStart + xEnd);
       }
 
+      if (model.discretize) {
+        const range = model.mappingRange;
+        if (x > range[0] && x < range[1]) {
+          const numberOfValues = model.numberOfValues;
+          const deltaRange = range[1] - range[0];
+          if (numberOfValues <= 1) {
+            x = range[0] + deltaRange / 2.0;
```

Comment:
if numberOfValues == 1 then you should have the same color for x = range[0], x = range[1] and x = (range[0]+range[1]) /2)
here you will have 3 colors (because of your `if` condition line 499)